### PR TITLE
Bak 1063/improve lookups

### PR DESCRIPTION
**Pull request by THOR300**
***Work in Progress PR*** 

Improve design for lookups - BAK/1063

- Added a new backend endpoint "/config"
- Integrated tests for the endpoint (these successfully pass on the make test_backend command)  
- Integrated response_model for the endpoint 
- Updated the frontend to use the new endpoint and render the response data

### CODE REVIEW

**Review by THOR300 — COMMENTED** [review]
**/.idea should ignore in all folders (globbing)

File: frontend/src/pages/search/index.tsx [hunk 1]
```diff
@@ -62,20 +63,20 @@ const Search = () => {
...
   // get lookups/filters
-  const documentTypesQuery: any = useLookups("document_types");
-  const { data: { data: documentTypes = {} } = {} } = documentTypesQuery;
-
-  const geosQuery: any = useNestedLookups("geographies", "", 2);
-  const { data: { data: { level1: regions = [], level2: countries = [] } = {} } = {} } = geosQuery;
+  const configQuery: any = useConfig("config");
+  const {
+            data: {
+                    document_types: document_types = {},
```
Comment (PatFawbertMills) [inline frontend/src/pages/search/index.tsx:69]: These collections need to default to empty arrays `[]` as opposed to objects.

File: frontend/src/hooks/useConfig.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        geographies,
+        instruments,
+        sectors,
+        regions,
+        countries,
+      }
+      return resp_end;
+    },
+    {
+      onSuccess: (data) => {
```
Comment (PatFawbertMills) [inline frontend/src/hooks/useConfig.ts:57]: Can remove this as it isn't doing anything and we don't need to fire off any side-affects from the query

File: frontend/src/hooks/useLookups.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default function useLookups(path: string, filterProp: string = '') {
     },
     {
       onSuccess: (data) => {
```
Comment (PatFawbertMills) [inline frontend/src/hooks/useLookups.ts:24]: Same as `useConfig` we can remove this

**Review by PatFawbertMills — CHANGES_REQUESTED** [review]
Just need to remove a couple of the query `onSuccess` methods that we don't use

**Review by PatFawbertMills — APPROVED** [review]
Looks good nice one 👍

**Pull request merged**